clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1032,18 +1032,13 @@ private ClusterAndWaitTime waitOnMetadata(String topic, Integer partition, long
                 metadata.awaitUpdate(version, remainingWaitMs);
             } catch (TimeoutException ex) {
                 // Rethrow with original maxWaitMs to prevent logging exception with remainingWaitMs
-                throw new TimeoutException(
-                        String.format("Topic %s not present in metadata after %d ms.",
-                                topic, maxWaitMs));
+                throw timeoutForAbsentMetadata(topic, partition, maxWaitMs, partitionsCount);
```

Comment:
Is it actually possible for this to be thrown when the partition is missing? We only pass the topic to `metadata.requestUpdateForTopic`.